### PR TITLE
Tilde extension to allow replacing plugin modules

### PR DESCRIPTION
**Pull request by justinbmeyer**
This implements the tilde extension `src/system-extension-tilde.js` which makes it easy for plugins to define places that can be replaced by `normalized` and `located` modules.  The idea is that a plugin can identify a regular expression (or an array of regular expressions) which will be used to replace moduleNames located in those regular expressions with a full address which has been found by running the module name through normalize and locate hook.

This applies replacements for less and css as outlined in issue #277

### CODE REVIEW

File: test/ext/css.js [hunk 1]
```diff
@@ -70,6 +70,6 @@ function isProduction(){
 	return (loader.isEnv && loader.isEnv("production")) ||
 		loader.env === "production";
 }
-
```
Comment (matthewp) [inline test/ext/css.js]: didn't you take this option away?

File: src/system-extension-tilde.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
+var addTilde = function(loader){
```
Comment (matthewp) [inline src/system-extension-tilde.js]: can we rename this to something else, `system-plugin-locate.js` perhaps?

**Comment by matthewp** [comment]
I'm wonder if this should be a configDependency rather than in core. I'm not sure about adding 120LOC for such a specific use-case, seems like it would make a nice configDep.

**Comment by matthewp** [comment]
What would be even nicer is if we had a proper extension system as described in https://github.com/stealjs/steal/issues/414 and the [StealJS roadmap](http://forums.donejs.com/t/stealjs-roadmap-discussion/131), that way this could be pushed to NPM. But unfortunately for now moving this into `ext/` and adding it as a configDep is probably the way to go.

**Comment by matthewp** [comment]
This is currently merged into minor (along with progressive package.jsons) so if you need to do any more prerelease let me know.

**Comment by pYr0x** [comment]
to be clear, this PR is replaced by https://github.com/stealjs/steal/issues/605 ?

**Comment by matthewp** [comment]
That's right.

**Pull request merged**